### ISSUE TITLE
junit_test is incompatible with JDK 18+ - "java.lang.UnsupportedOperationException: The Security Manager is deprecated"

### ISSUE THREAD

**Issue by sitaktif** [bazelbuild/bazel#16421]
### Description of the bug:

Running a trivial `java_test` with JDK 18+ yields the following (with the test failing):

```
JUnit4 Test Runner
java.lang.UnsupportedOperationException: The Security Manager is deprecated and will be removed in a future release
	at java.base/java.lang.System.setSecurityManager(System.java:418)
	at com.google.testing.junit.runner.junit4.JUnit4Runner.installSecurityManager(JUnit4Runner.java:256)
	at com.google.testing.junit.runner.junit4.JUnit4Runner.run(JUnit4Runner.java:113)
	at com.google.testing.junit.runner.BazelTestRunner.runTestsInSuite(BazelTestRunner.java:148)
	at com.google.testing.junit.runner.BazelTestRunner.main(BazelTestRunner.java:75)

BazelTestRunner exiting with a return value of 1
```

Having JDK 17 (or older) on the PATH fixes the issue above.

#### Sources

`WORKSPACE.bazel`:

```
(Empty)
```

`java/example/BUILD.bazel`:

```python
java_test(
    name = "ExampleTest",
    srcs = ["ExampleTest.java"],
)
```

`java/example/ExampleTest.java`:

```java
package example;

import org.junit.Test;

public class ExampleTest {
  @Test
  public void passes() {
  }
}
```

#### Repro command 

```bash
$ java --version
openjdk 18.0.1
[...]

$ bazel test ... --test_output=errors
[...]
JUnit4 Test Runner
java.lang.UnsupportedOperationException: The Security Manager is deprecated and will be removed in a future release
	at java.base/java.lang.System.setSecurityManager(System.java:418)
	at com.google.testing.junit.runner.junit4.JUnit4Runner.installSecurityManager(JUnit4Runner.java:256)
	at com.google.testing.junit.runner.junit4.JUnit4Runner.run(JUnit4Runner.java:113)
	at com.google.testing.junit.runner.BazelTestRunner.runTestsInSuite(BazelTestRunner.java:148)
	at com.google.testing.junit.runner.BazelTestRunner.main(BazelTestRunner.java:75)

BazelTestRunner exiting with a return value of 1
[...]
INFO: Build completed, 1 test FAILED
[...]
```

### Which operating system are you running Bazel on?=

Macos 12.6

### What is the output of `bazel info release`?

release 5.1.1

**Comment by fmeum** [comment]
You can specify `--jvmopt=-Djava.security.manager=allow` on the command-line or add `common --jvmopt=-Djava.security.manager=allow` as a workaround that allows you to use JDK 18.

**Comment by fmeum** [comment]
@sgowroji This is essentially a duplicate of https://github.com/bazelbuild/bazel/issues/14502, but, in my opinion, a real bug - `java_test` should work with any released JDK by default.

**Comment by sgowroji** [comment]
Agreed, Thanks for sharing the workaround and marking it to Java team.

**Comment by comius** [comment]
Closing as duplicate of https://github.com/bazelbuild/bazel/issues/14502

**Comment by toandaominh1997** [comment]
@fmeum This is a great way to fix Security Manager is deprecated on JDK 18

**Comment by peakschris** [comment]
I note that this issue is closed, but with Bazel 7.3.1, and jdk 17, and `common --jvmopt=-Djava.security.manager=allow`, I am still seeing this issue when running junit tests via bazel test:

```
WARNING: A terminally deprecated method in java.lang.System has been called
WARNING: System::setSecurityManager has been called by com.google.testing.junit.runner.util.GoogleTestSecurityManager (file:/D:/udu/b/xrcdxqdk/external/rules_java++toolchains+remote_java_tools/java_tools/Runner_deploy.jar)
WARNING: Please consider reporting this to the maintainers of com.google.testing.junit.runner.util.GoogleTestSecurityManager
WARNING: System::setSecurityManager will be removed in a future release
```
